### ISSUE TITLE
Integrated AWS API Gateway Event configuration

### ISSUE THREAD

**Issue by flomotlik** [serverless/serverless#1164]
### Necessary resources:
1. AWS::ApiGateway::RestApi (create rest api)
2. AWS::ApiGateway::Deployment (deploy rest api and expose to internet)
3. AWS::ApiGateway::Stage (for each stage)
5. AWS::ApiGateway::Resource (resource for the method)
6. AWS::ApiGateway::Method (most complex resource probably)
7. AWS::Lambda::Permission (give APIG permission to invoke lambda)

### REST API name prefixing
`restApi` name must be prefixed with stage (e.g. `dev-foo`)

**Comment by HyperBrain** [comment]
It would be good to have the API event configuration separated by `method` or `request` to have no conflicts with the HTTP method and `integration` like this:

#### serverless.yml
```
functions:
  myfunc:
    events:
      - my_endpoint:
          path: /v1/my/endpoint
          method: post
          type: HTTP | MOCK | FUNCTION
          request:
            yyyyy
          integration:
            xxxxx
```

This shall solve the current problem that the method and integration settings are mixed up.
Integration should be optional to have support for the MOCK type.

**Comment by pmuens** [comment]
Necessary resources:

1. AWS::ApiGateway::RestApi (create rest api)
2. AWS::ApiGateway::Deployment (deploy rest api and expose to internet)
3. AWS::ApiGateway::Stage (for each stage)
4. AWS::ApiGateway::BasePathMapping (for each stage. ie example.com/dev/<endpoint>)
5. AWS::ApiGateway::Resource (resource for the method)
6. AWS::ApiGateway::Method (most complex resource probably)
7. AWS::Lambda::Permission (give APIG permission to invoke lambda)

**Comment by pmuens** [comment]
Resolved with #1312
